zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/SnapStream.java

```diff
@@ -0,0 +1,336 @@
+        oa.writeLong(val, "val");
+        oa.writeString("/", "path");
+    }
+
+    /**
+     * Verify the integrity of the seal, only CheckedInputStream will verify
+     * the checkSum of the content.
+     *
+     */
+    static void checkSealIntegrity(CheckedInputStream is, InputArchive ia)
+            throws IOException {
+        long checkSum = is.getChecksum().getValue();
+        long val = ia.readLong("val");
+        if (val != checkSum) {
+            throw new IOException("CRC corruption");
+        }
+    }
+
+    /**
+     * Verifies that the file is a valid snapshot. Snapshot may be invalid if
+     * it's incomplete as in a situation when the server dies while in the
+     * process of storing a snapshot. Any files that are improperly formated
+     * or corrupted are invalid. Any file that is not a snapshot is also an
+     * invalid snapshot.
+     *
+     * @param file file to verify
+     * @return true if the snapshot is valid
+     * @throws IOException
+     */
+    public static boolean isValidSnapshot(File file) throws IOException {
+        if (file == null || Util.getZxidFromName(file.getName(), FileSnap.SNAPSHOT_FILE_PREFIX) == -1) {
+            return false;
+        }
+
+        String fileName = file.getName();
+        if (Util.getZxidFromName(fileName, "snapshot") == -1) {
+            return false;
+        }
+
+        boolean isValid = false;
+        switch (getStreamMode(fileName)) {
+            case GZIP:
+                isValid = isValidGZipStream(file);
+                break;
+            case SNAPPY:
+                isValid = isValidSnappyStream(file);
+                break;
+            case CHECKED:
+            default:
+                isValid = isValidCheckedStream(file);
+        }
+        return isValid;
+    }
+
+    public static void setStreamMode(StreamMode mode) {
+        streamMode = mode;
+    }
+
+    public static StreamMode getStreamMode() {
+        return streamMode;
+    }
+
+    /**
+     * Detect the stream mode from file name extension
+     *
+     * @param fileName
+     * @return
+     */
+    public static StreamMode getStreamMode(String fileName) {
+        String[] splitSnapName = fileName.split("\\.");
+
+        // Use file extension to detect format
+        if (splitSnapName.length > 1) {
+            String mode = splitSnapName[splitSnapName.length - 1];
+            return StreamMode.fromString(mode);
+        }
+
+        return StreamMode.CHECKED;
+    }
+
+    /**
+     * Certify the GZip stream integrity by checking the header
+     * for the GZip magic string
+     *
+     * @param f file to verify
+     * @return true if it has the correct GZip magic string
+     * @throws IOException
+     */
+    private static boolean isValidGZipStream(File f) throws IOException {
+        FileInputStream fis = null;
+        byte[] byteArray = new byte[2];
+
+        try {
+            fis = new FileInputStream(f);
+            fis.read(byteArray, 0, 2);
```

Comment:
Findbugs reports this, as you ignore the result of read().
Please add it to findbugsExcludeFile.xml to ignore it (or you can use return value to check if the read was successful here, not necessary though).

Reply:
See https://builds.apache.org/job/PreCommit-ZOOKEEPER-github-pr-build/2623/artifact/build/test/findbugs/newPatchFindbugsWarnings.html

Reply:
This is actually a bug to be fixed.
You can for instance use DataInputStream#readFully or use IOUtils if we have it on the classpath

Is this case it is enough to fail if the result is != 2

Reply:
We can also use try-with-resources

Reply:
Fixed Findbugs warnings and changed to using try-with-resources!